### PR TITLE
Deprecate usage of inject_patch.

### PR DESCRIPTION
**Pull request by andresito00**
**Link to Related Issue(s)**
https://github.com/redballoonsecurity/ofrak/pull/122

**Please describe the changes in your request.**
Replace usage of inject_patch with SegmentInjectorModifier

**Anyone you think should look at this, specifically?**
@EdwardLarson @whyitfor

### CODE REVIEW

File: ofrak_patch_maker/ofrak_patch_maker/patch_maker.py [hunk 1]
```diff
@@ -28,7 +28,8 @@
 
 fem = patch_maker.make_fem([(bom, region_config)], ofrak_fw_resource, verbose=True)
 
-patch_maker.inject_patch(verbose=True)
+# deprecated patch_maker.inject_patch(fem, ofrak_fw_resource, verbose=True)
+ofrak_fw_resource.run(SegmentInjectorModifier, SegmentInjectorModifierConfig.from_fem(fem))
```
Comment (EdwardLarson) [inline ofrak_patch_maker/ofrak_patch_maker/patch_maker.py:32]:
```suggestion
await ofrak_fw_resource.run(SegmentInjectorModifier, SegmentInjectorModifierConfig.from_fem(fem))
```

File: ofrak_patch_maker/ofrak_patch_maker/patch_maker.py [hunk 2]
```diff
@@ -574,6 +575,7 @@ async def inject_patch(
         :param resource: OFRAK firmware resource
         :param verbose: prints patch section sizes and destinations at the effective logging level.
         """
+        self.logger.warn("This function is being deprecated!!")
```
Comment (whyitfor) [inline ofrak_patch_maker/ofrak_patch_maker/patch_maker.py:578]: I'd be comfortable just deleting this method.

If you want to keep it around during the transition, then I'd suggest moving the ofrak imports into the body of the method (and making `resource: "Resource"` typehint in signature.

Comment (EdwardLarson) [reply ofrak_patch_maker/ofrak_patch_maker/patch_maker.py:578]: +1, I don't love keeping this method around but I can see how it helps the transition. I really like the idea of moving the imports into the body of the function.

Comment (andresito00) [reply ofrak_patch_maker/ofrak_patch_maker/patch_maker.py]: Done.

**Review by EdwardLarson — APPROVED** [review]

**Pull request merged**